google-cloud-storage/src/test/java/com/google/cloud/storage/conformance/retry/State.java

```diff
@@ -289,6 +293,10 @@ public <T> State consume(Page<T> page) {
     return newStateWith(KEY_LIST_OBJECTS, collect);
   }
 
+  public <T> State with(List<T> listResult) {
+    return newStateWith(KEY_LIST_OBJECTS, listResult);
```

Comment:
Can we add a new `KEY_LIST_NOTIFICATION` (and corresponding `has`, `get` and `set` methods) so that we don't possibly overwrite objects that might already be held?

```
private static final Key<List<Notification>> KEY_LIST_NOTIFICATION = new Key<>("list<notification>");
```